test/test_helper.rb

```diff
@@ -131,66 +131,61 @@ def self.create(problem)
     Models::Vrp.create(coerce(problem))
   end
 
-  def self.matrix_required(vrp)
-    return unless ENV['TEST_DUMP_VRP'].to_s == 'true' && vrp.name
+  def self.matrices_required(vrps, filename)
+    return if vrps.all?{ |vrp| vrp.matrices.any? }
+    
+    dump_file = "test/fixtures/#{filename}.dump"
+    dumped_data = Marshal.load(Zlib.inflate(File.read(dump_file))) if File.file?(dump_file) # rubocop: disable Security/MarshalLoad
 
-    WebMock.enable_net_connect!
-    vrp.compute_matrix
-    WebMock.disable_net_connect!
-    File.write("test/fixtures/#{vrp.name}.dump", Zlib.deflate(Marshal.dump(vrp)))
-  end
+    write_in_dump = []
+    vrps.each{ |vrp|
+      next if vrp.matrices.any?
 
-  def self.multipe_matrices_required(vrps, filename)
-    return unless ENV['TEST_DUMP_VRP'].to_s == 'true'
+      if dumped_data.nil? && ENV['TEST_DUMP_VRP'] != 'true'
+        WebMock.enable_net_connect!
+        vrp.compute_matrix
+        WebMock.disable_net_connect!
+      else
+        Routers::RouterWrapper.stub_any_instance(:matrix, lambda { |url, mode, dimensions, row, column, options|
+          if ENV['TEST_DUMP_VRP'] == 'true'
+            matrices =
+              Routers::RouterWrapper.send(:__minitest_stub__matrix, url, mode, dimensions, row, column, options) # call original method
+            write_in_dump <<
+              { url: url, mode: mode, dimensions: dimensions, row: row, column: column, options: options, matrices: matrices }
+            matrices
+          else
+            corresponding_matrices = dumped_data.find{ |request|
+              # TODO : the day filter.rb (from models/vrp.rb) filters services then some test may not find
+              # their corresponding matrix anymore because row and/or column will be different
+              request[:mode] == mode && request[:dimensions] == dimensions &&
+                request[:row] == row && request[:column] == column && request[:options] == options
+            }[:matrices]
+            dimensions.collect.with_index{ |_dim, dim_i| corresponding_matrices[dim_i] }
```

Comment:
```suggestion
              request[:mode] == mode && request[:dimensions] == dimensions &&
                request[:options] == options && request[:row] == row && request[:column] == column
            }[:matrices]
            # TODO: If services are filtered then some test may not find
            # their corresponding matrices because row and/or column will be different
            # In this case, we need to modify the above code and filter via lat/lon
            # to have the same order/content.
            raise 'Could not find matrix in the dump' if corresponding_matrices.nil?
            dimensions.collect.with_index{ |_dim, dim_i| corresponding_matrices[dim_i] }
```

Reply:
I added a raise, moved the find condition order and modified the comment